assets/src/edit-story/app/uploader/test/useUploader.js

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { renderHook } from '@testing-library/react-hooks';
+import { render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import ConfigContext from '../../config/context';
+import useUploader from '../useUploader';
+
+function setup(args, refreshLibrary = false) {
+  const configValue = {
+    api: {},
+    allowedMimeTypes: {
+      image: ['image/png', 'image/jpeg', 'image/jpg', 'image/gif'],
+      video: ['video/mp4'],
+    },
+    allowedFileTypes: ['png', 'jpeg', 'jpg', 'gif', 'mp4'],
+    maxUpload: 104857600,
+    capabilities: {
+      hasUploadMediaAction: true,
+    },
+    ...args,
+  };
+  const wrapper = (params) => (
+    <ConfigContext.Provider value={configValue}>
+      {params.children}
+    </ConfigContext.Provider>
+  );
+  const { result } = renderHook(() => useUploader(refreshLibrary), { wrapper });
+  return {
+    uploadFile: result.current.uploadFile,
+    isValidType: result.current.isValidType,
+  };
+}
+
+describe('useUploader', () => {
+  it('user unable to upload', () => {
+    const { uploadFile } = setup({
+      capabilities: {
+        hasUploadMediaAction: false,
+      },
+    });
+    try {
+      uploadFile({});
+    } catch (e) {
+      // eslint-disable-next-line jest/no-try-expect
+      expect(e.message).toStrictEqual('Sorry, you are unable to upload files.');
+    }
+  });
+  it('user uploads a to large file', () => {
+    const { uploadFile } = setup({
+      maxUpload: 2000000,
+    });
+    try {
+      uploadFile({ size: 3000000 });
+    } catch (e) {
+      // eslint-disable-next-line jest/no-try-expect
+      expect(e.message).toStrictEqual(
+        'Your file is 3MB and the upload limit is 2MB. Please resize and try again!'
+      );
+    }
+  });
+  it('user uploads an inavlid file', () => {
+    const { uploadFile } = setup({});
+    try {
+      uploadFile({ size: 20000, type: 'application/pdf' });
+    } catch (e) {
+      const { asFragment } = render(e.message);
+      // eslint-disable-next-line jest/no-try-expect
```

Comment:
See https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-try-expect.md for the recommended way to test this instead of disabling this rule.